zorkatest.js

```diff
@@ -0,0 +1,417 @@
+// prebuilt code
+const readline = require("readline");
+const { start } = require("repl");
+const rlInterface = readline.createInterface(process.stdin, process.stdout);
+
+function ask(questionText) {
+  return new Promise((resolve, reject) => {
+    rlInterface.question(questionText, resolve);
+  });
+}
+//start of my code
+class Item {
+  constructor(iName, iDescription, iAction, iTakeable, Idurability) {
+    this.name = iName;
+    this.description = iDescription;
+    this.takeable = iTakeable || false;
+    this.action = iAction || "nothing happens";
+    this.durability = Idurability;
+  }
+  take() {
+    //check if item takable and is in room's inventory
+    if (this.takeable && locationTable[curLoc].inventory.includes(this.name)) {
+      //add item to player inventory
+      playerInventory.push(this.name);
+      //remove item from the room's inventory
+      let takeIndex = locationTable[curLoc].inventory.indexOf(this.name);
+      locationTable[curLoc].inventory.splice(takeIndex, 1);
+      //return a confirmation value to user
+      return `you picked up ${this.name}`;
+    } else {
+      return `you are unable to pick that up`;
+    }
+  }
+  use() {
+    //console.log(this.durability);
```

Comment:
there are several console.logs in your code that are commented out but should be deleted.